aquadoggo/src/graphql/replication/mod.rs

```diff
@@ -184,84 +182,106 @@ mod tests {
-            "
-        query{{
-          getEntriesNewerThanSeq(logId: {}, author: {{publicKey: \"{}\" }}, sequenceNumber:{}, first: {}, after: \"{}\" ){{
-            pageInfo {{
-              hasNextPage
-            }}
-          }}
-        }}",
-            log_id, author_string, sequence_number, first, after.encode_cursor()
-        );
-
-        let mut replication_context: ReplicationContext<MockEntryStore> =
-            ReplicationContext::default();
-
-        // Prepare our main assertions.
-        // - Checks that get_entries_newer_than_seq is called with the values we expect
-        // - Checks that get_entries_newer_than_seq is called once
-        // - Configures get_entries_newer_than_seq to return an empty Vec
-        replication_context
-            .expect_get_entries_newer_than_seq()
-            .withf({
-                let author_string = author_string.clone();
-
-                move |log_id_, author_, sequence_number_, max_number_of_entries_| {
-                    let author_matches = match author_ {
-                        AuthorOrAlias::PublicKey(public_key) => {
-                            public_key.0.as_str() == author_string
-                        }
-                        _ => false,
-                    };
-                    sequence_number_.as_u64() == expected_start
-                        && log_id_.as_u64() == log_id
-                        && author_matches
-                        && *max_number_of_entries_ == first
-                }
-            })
-            .returning(|_, _, _, _| Ok(vec![]))
-            .once();
-
-        // Build up a schema with our mocks that can handle gql query strings
-        let replication_root = ReplicationRoot::<MockEntryStore>::new();
-        let schema = Schema::build(replication_root, EmptyMutation, EmptySubscription)
-            .data(Arc::new(Mutex::new(replication_context)))
-            .finish();
-
-        // Act
-        let result = schema.execute(Request::new(gql_query)).await;
-
-        // Assert
-
-        // Check that we get the Ok returned from get_entries_newer_than_seq
-        assert!(result.is_ok());
-
-        // There should not be a next page because we returned an empty vec from
-        // get_entries_newer_than_seq and it's length is not == `first` == 5;
-        let json_value = result.data.into_json().unwrap();
-        let has_next_page = &json_value["getEntriesNewerThanSeq"]["pageInfo"]["hasNextPage"];
-        assert!(!has_next_page.as_bool().unwrap());
+    use crate::db::stores::test_utils::{
+        populate_test_db, with_db_manager_teardown, PopulateDatabaseConfig, TestDatabaseManager,
+    };
+    use crate::SqlStorage;
+
+    use super::{ReplicationContext, ReplicationRoot};
+
+    #[rstest]
+    #[case(100, None, None, None, true, 10)]
+    #[case(10, Some(10), Some(5), None, false, 0)]
+    #[case(14, Some(10), Some(5), None, false, 4)]
+    #[case(15, Some(10), Some(5), None, true, 5)]
```

Comment:
I was surprised that this case returned `hasNextPage = true` is that what we would expect?

Reply:
Oh ja, thats also not what I'd expect